glib/src/source.rs

```diff
@@ -142,6 +142,56 @@ fn into_raw_unix_fd<F: FnMut(RawFd, IOCondition) -> Continue + 'static>(func: F)
     Box::into_raw(func) as gpointer
 }
 
+///
+/// In comparisong to `timeout_add()`, this only requires `func` to be
+/// `FnOnce`, and will automatically return `Continue(false)`.
+where
+    F: FnOnce() + Send + 'static,
+
+
+/// Transform a generic FnOnce into a closure that can be used as callback in various glib methods
+///
+/// The resulting function can only be called once and will panic otherwise. It will return `Continue(false)`
+/// in order to prevent being called twice.
+#[inline(always)]
+fn fnmut_callback_wrapper(func: F) -> G
+where
+    F: FnOnce() + Send + 'static,
+    G: FnMut() -> Continue + Send + 'static,
+{
+    let mut func = Some(func);
+    move || {
+        func.take().unwrap();
```

Comment:
```suggestion
        func.take().expect("GSource closure called after returning glib::Continue(false)");
```

Or something like that